src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -120,12 +123,18 @@ export class ClusterClient implements IClusterClient {
    */
   private isClosed = false;
 
+  private init: Promise<void>;
+  private scopedInit?: Promise<void>;
+
   constructor(
     private readonly config: ElasticsearchClientConfig,
     private readonly log: Logger,
     private readonly getAuthHeaders: GetAuthHeaders = noop
   ) {
-    this.client = new Client(parseElasticsearchClientConfig(config, log));
+    this.init = new Promise(async resolve => {
+      this.client = new Client(await parseElasticsearchClientConfig(this.config, this.log));
+      resolve();
+    });
```

Comment:
Parsing the Elasticsearch client config used to be synchonous, but now it is asynchronous. However, the ClusterClient is only used in an asynchronous manner -- we can create a Promise which will eager-load the Elasticsearch client config, and then only allow any other calls to proceed after initialization is complete.
A single promise will only call its executor function one time, and after it resolves it will return the same result every time it is used.